libs/neo4j/langchain_neo4j/vectorstores/neo4j_vector.py

```
@@ -778,60 +419,59 @@ def retrieve_existing_fts_index(
         Returns:
             (Tuple): keyword index information
         """
-
-        index_information = self.query(
-            "SHOW INDEXES YIELD name, type, labelsOrTypes, properties, options "
-            "WHERE type = 'FULLTEXT' AND (name = $keyword_index_name "
-            "OR (labelsOrTypes = [$node_label] AND "
-            "properties = $text_node_property)) "
-            "RETURN name, labelsOrTypes, properties, options ",
-            params={
-                "keyword_index_name": self.keyword_index_name,
-                "node_label": self.node_label,
-                "text_node_property": text_node_properties or [self.text_node_property],
-            },
-        )
-        # sort by index_name
-        index_information = sort_by_index_name(index_information, self.index_name)
-        try:
-            self.keyword_index_name = index_information[0]["name"]
-            self.text_node_property = index_information[0]["properties"][0]
-            node_label = index_information[0]["labelsOrTypes"][0]
-            return node_label
-        except IndexError:
+        if self.keyword_index_name:
+            index_information = retrieve_fulltext_index_info(
+                driver=self._driver,
+                index_name=self.keyword_index_name,
+                label_or_type=self.node_label,
+                text_properties=text_node_properties or [self.text_node_property],
+            )
+        else:
+            raise ValueError("keyword_index_name is not set.")
+        if index_information:
+            try:
+                self.keyword_index_name = index_information["name"]
+                self.text_node_property = index_information["properties"][0]
+                node_label = index_information["labelsOrTypes"][0]
+                return node_label
+            except IndexError:
+                return None
```

Same here regarding silent failure